lib/Doctrine/DBAL/Query/Expression/CompositeExpression.php

```diff
@@ -91,7 +91,7 @@ public function addMultiple(array $parts = [])
      */
     public function add($part)
     {
-        if ( ! empty($part) || ($part instanceof self && $part->count() > 0)) {
+        if (!empty($part) && count($part)) {
```

Comment:
What if `$part` is not empty but not countable? E.g. `$expr->add('id = 1')`. It will fail on PHP 7.2.

See https://wiki.php.net/rfc/counting_non_countables